src/store/modules/activity_daily.ts

```diff
@@ -93,102 +97,128 @@ const getters = {
 
 // actions
 const actions = {
-  async ensure_loaded({ commit, state, dispatch }, query_options: QueryOptions) {
+  async ensure_loaded({ commit, state, rootState, dispatch }, query_options: QueryOptions) {
     console.info('Query options: ', query_options);
     if (!state.loaded || state.query_options !== query_options || query_options.force) {
       commit('start_loading', query_options);
       if (!query_options.timeperiod) {
         query_options.timeperiod = dateToTimeperiod(query_options.date);
       }
+
+      await dispatch('buckets/ensureBuckets', null, { root: true });
       await dispatch('get_buckets', query_options);
 
       // TODO: These queries can actually run in parallel, but since server won't process them in parallel anyway we won't.
       await dispatch('query_window', query_options);
       await dispatch('query_browser', query_options);
-      await dispatch('query_editor', query_options);
       await dispatch('query_active_history', query_options);
+      await dispatch('query_editor', query_options);
     } else {
       console.warn(
         'ensure_loaded called twice with same query_options but without query_options.force = true, skipping...'
       );
     }
   },
 
-  async query_window({ commit }, { host, timeperiod, filterAFK, filterCategories }: QueryOptions) {
-    const start = moment();
-    const periods = [timeperiodToStr(timeperiod)];
-    const classes = loadClassesForQuery();
-    const bucket_id_window = 'aw-watcher-window_' + host;
-    const bucket_id_afk = 'aw-watcher-afk_' + host;
-    const q = queries.windowQuery(
-      bucket_id_window,
-      bucket_id_afk,
-      default_limit, // this.top_apps_count,
-      default_limit, // this.top_windowtitles_count,
-      filterAFK,
-      classes,
-      filterCategories
-    );
-    const data = await this._vm.$aw.query(periods, q);
-    console.info(`Completed window query in ${moment().diff(start)}ms`);
-    commit('query_window_completed', data[0]);
+  async query_window({ state, commit }, { host, timeperiod, filterAFK, filterCategories }: QueryOptions) {
+    if (state.buckets.afk_buckets.length > 0 &&
+        state.buckets.window_buckets.length > 0) {
```

Comment:
Not really a fan of this check being here. Bucket-requirements should imo be checked in `ensure_loaded` and the `start_loading` thing should be called in the individual queries instead of one-fits-all.

Not a blocker for this PR, but would be nice if you could add that as a TODO comment in the code.

Reply:
> Not really a fan of this check being here. Bucket-requirements should imo be checked in ensure_loaded

Done

>  the start_loading thing should be called in the individual queries instead of one-fits-all.

Not sure what you mean by this?

Reply:
Meant exactly what you changed it to, nice!